File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        if(n == 0) return {0};
        vector<int> ans{0,1};

        function<int(unsigned int)> countOnes = [&](unsigned int i) -> int {
            int count = 0;
            while(i){
                i &= i-1;
                ++count;
            }
            return count;
        };

        for(unsigned int i=2; i<n+1; ++i){
            ans.push_back(countOnes(i));
        }

        return ans;
    }
};

File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        def help(n: int) -> int:
            cnt = 0
            for i in range(0,32): 
                if (n >> i) & 1: cnt += 1
            return cnt
        answer = [ help(i) for i in range(0,n+1) ]
        return answer